lib/grape_entity/entity.rb

```diff
@@ -499,8 +499,8 @@ def should_return_attribute?(attribute, options)
       only_fields(options).include?(self.class.key_for(attribute))
     end
 
-    def only_fields(options)
-      return {} unless options[:only]
+    def only_fields(options, for_attribute = nil)
+      return nil unless options[:only]
```

Comment:
Pedantic change, you don't need to `return nil`, just `return`, `nil` is by default.